openverse_catalog/dags/providers/factory_utils.py

```diff
@@ -0,0 +1,137 @@
+        # class, which will initialize the media stores and DelayedRequester.
+        logger.info(
+            f"Initializing ProviderIngester {ingestion_callable.__name__} in"
+            f"order to generate store filenames."
+        )
+        ingester = ingestion_callable(*args)
+        stores = ingester.media_stores
+
+    # Push the media store output paths to XComs.
+    for store in stores.values():
+        logger.info(
+            f"{store.media_type.capitalize()} store location: {store.output_path}"
+        )
+        ti.xcom_push(key=f"{store.media_type}_tsv", value=store.output_path)
+
+
+def pull_media_wrapper(
+    ingestion_callable: Callable,
+    media_types: list[MediaType],
+    tsv_filenames: list[str],
+    ti: TaskInstance,
+    args: Sequence = None,
+):
+    """
+    Run the provided callable after pushing setting the output directories
+    for each media store using the provided values. This is a temporary workaround
+    due to the nature of the current provider scripts. Once
+    https://github.com/WordPress/openverse-catalog/issues/229 is addressed and the
+    provider scripts are refactored into classes, this wrapper can either be updated
+    or the output filename retrieval/setting process can be altered.
+    """
+    args = args or []
+    if len(media_types) != len(tsv_filenames):
+        raise ValueError(
+            "Provided media types and TSV filenames don't match: "
+            f"{media_types=} {tsv_filenames=}"
+        )
+    logger.info("Setting media stores to the appropriate output filenames")
+
+    # TODO: This entire branch can be removed when all of the provider scripts have been
+    # TODO: refactored to subclass ProviderDataIngester.
+    if isinstance(ingestion_callable, FunctionType):
+        # Stores exist at the module level, so in order to set the output values we
+        # must first pull the stores from the module.
+        stores = _load_provider_script_stores(ingestion_callable, media_types)
+        run_func = ingestion_callable
+    else:
+        # A ProviderDataIngester class was passed instead. First we initialize the
+        # class, which will initialize the media stores and DelayedRequester.
+        logger.info(f"Initializing ProviderIngester {ingestion_callable.__name__}")
+        ingester = ingestion_callable(*args)
+        stores = ingester.media_stores
+        run_func = ingester.ingest_records
+
+    for store, tsv_filename in zip(stores.values(), tsv_filenames):
+        logger.info(
+            f"Setting {store.media_type.capitalize()} store location "
+            f"to {tsv_filename}"
+        )
+        store.output_path = tsv_filename
+
+    logger.info("Beginning ingestion")
+    start_time = time.perf_counter()
+    # Not passing kwargs here because Airflow throws a bunch of stuff in there that
+    # none of our provider scripts are expecting.
+    data = run_func(*args)
```

Comment:
This is unfortunate because I love how nice the refactor here looks -- but while we *do* need to pass `*args` here for legacy scripts, we don't actually want to pass them in here if we're using an ingester class.

For those, we pass `*args` to the `__init__` but we don't want to pass them to `ingest_records` as well, or else it causes problems with dated DAGs when the `date` is unexpectedly passed in (I found/fixed this separately in #618).

Reply:
Ah, okay I was a little thrown off when I first saw that! I was only copying what was present in the original code: https://github.com/WordPress/openverse-catalog/blob/cd7ca961c169320af1a3c69c514d880b9ac0dd85/openverse_catalog/dags/providers/provider_dag_factory.py#L138-L150

I think I should be able to adjust things here to make that case work!

Reply:
Yes, exactly! I meant I noticed it on main as well when working on #618, but that's still in draft. It's wasn't found before because none of the dated DAGs have been refactored yet. I like your fix!